jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketCoreSession.java

```diff
@@ -144,15 +145,41 @@ public void setIdleTimeout(Duration timeout)
     @Override
     public Duration getWriteTimeout()
     {
-        return writeTimeout;
+        return getFrameWriteTimeout();
+    }
+
+    @Override
+    public Duration getFrameWriteTimeout()
+    {
+        return frameWriteTimeout;
+    }
+
+    @Override
+    public Duration getMessageWriteTimeout()
+    {
+        return messageWriteTimeout;
     }
 
     @Override
     public void setWriteTimeout(Duration timeout)
     {
-        writeTimeout = timeout;
+        setFrameWriteTimeout(timeout);
+    }
+
+    @Override
+    public void setFrameWriteTimeout(Duration timeout)
+    {
+        frameWriteTimeout = timeout;
```

Comment:
```suggestion
        frameWriteTimeout = Objects.requireNonNull(timeout);
```

Below, also call `getConnection()` once and store it in a local variable.

Reply:
done